### PR TITLE
more verbose documentation regarding `.load` vs `.set_parameters` 

### LINKED ISSUE #614: Evalution error with BatchNorm
Hi! I am still faced with these problems since I use BatchNorm in my CustomCombinedExtractor. I have tried to modify some of the files with self.policy.eval(), but I still got unexpected test results. 
Are these problems fixed or not? I have reinstalled the codes under 'master', but it doesn't work. (The codes seem not changed as in the 'https://github.com/DLR-RM/stable-baselines3/pull/537/files')
Could you please offer some help? Thank you!

_Originally posted by @Leong1230 in https://github.com/DLR-RM/stable-baselines3/issues/537#issuecomment-945091297_

### LINKED ISSUE #683: [Bug] loading a trained MultiInput policy returns a random agent
### 🐛 Bug

I am using a MultiInput policy, trained with PPO. I am experiencing issues when loading a trained agent.
Specifically, the loaded model predicts actions differents from the original agent, even in deterministic mode.

### To Reproduce

A minimal example is reported below:
1. It trains PPO on CarRacing-v0 for a few steps, then saves and reloads the model.
1. The trained model and the loaded model are used to predict actions for the same observations in deterministic mode.
1. The actions are different as reported in the plot.

The code:
```python
import gym
import numpy as np
from stable_baselines3 import PPO
from stable_baselines3.common.env_checker import check_env

from racing_rl.envs.wrappers import FrameSkip
from skimage.color import rgb2gray


class VelocityWrapper(gym.ObservationWrapper):
    """ extend the observation space of racecar with velocity """

    def __init__(self, env):
        super(VelocityWrapper, self).__init__(env)
        self._w, self._h, c = self.observation_space.shape
        self.observation_space = gym.spaces.Dict({
            'image': gym.spaces.Box(low=0, high=255, shape=(80, 80, 1), dtype=np.uint8),
            'velocity': gym.spaces.Box(low=np.NINF, high=np.PINF, shape=(2,))
        })

    def observation(self, observation):
        image = np.reshape((rgb2gray(observation) * 255).astype(np.uint8), (self._w, self._h, 1))[:80, :80, :]
        image = np.where(image <= 150, 0, 255)
        dict_observation = {
            'image': image,
            'velocity': np.array(self.env.car.hull.linearVelocity)
        }
        return dict_observation


def twin_evaluations(model1, model2, env):
    """
    deterministic evaluation of 2 models when having the same observation as input.
    the action predicted by model1 is then used to interact with the environment.

    return a dictionary with the action for each model
    """
    done = False
    obs = env.reset()
    actions1, actions2 = [], []
    while not done:
        action1, _ = model1.predict(obs, deterministic=True)
        action2, _ = model2.predict(obs, deterministic=True)
        obs, reward, done, info = env.step(action1)
        actions1.append(action1)
        actions2.append(action2)
    return {'model1': np.array(actions1), 'model2': np.array(actions2)}


env = gym.make("CarRacing-v0")
env = VelocityWrapper(env)
env = FrameSkip(env, frame_skip=8)
check_env(env)
print("[info] env ok")

model = PPO("MultiInputPolicy", env, verbose=1)

model.learn(1000)
model.save("saved_model")

model2 = PPO("MultiInputPolicy", env, verbose=1, tensorboard_log=None)
model2.load("saved_model", print_system_info=True)

actions = twin_evaluations(model, model2, env)

import matplotlib.pyplot as plt

for i, name in enumerate(["steer", "gas", "brake"]):
    plt.subplot(3, 1, i + 1)
    plt.title(name)
    plt.plot(actions["model1"][:, i], label="after training")
    plt.plot(actions["model2"][:, i], label="after load")
    plt.legend()
plt.tight_layout()
plt.show()

```

The output:
```bash
/home/luigi/Development/racing-rl/venv/lib/python3.8/site-packages/stable_baselines3/common/env_checker.py:272: UserWarning: We recommend you to use a symmetric and normalized Box action space (range=[-1, 1]) cf https://stable-baselines3.readthedocs.io/en/master/guide/rl_tips.html
  warnings.warn(
Track generation: 1140..1429 -> 289-tiles track
[info] env ok
Using cuda device
Wrapping the env with a `Monitor` wrapper
Wrapping the env in a DummyVecEnv.
Wrapping the env in a VecTransposeImage.
Track generation: 1191..1493 -> 302-tiles track
Track generation: 1217..1525 -> 308-tiles track
Track generation: 1199..1503 -> 304-tiles track
Track generation: 926..1166 -> 240-tiles track
Track generation: 941..1186 -> 245-tiles track
Track generation: 1310..1641 -> 331-tiles track
retry to generate track (normal if there are not manyinstances of this message)
Track generation: 1324..1659 -> 335-tiles track
Track generation: 1320..1654 -> 334-tiles track
Track generation: 1107..1388 -> 281-tiles track
Track generation: 1062..1331 -> 269-tiles track
Track generation: 1033..1298 -> 265-tiles track
retry to generate track (normal if there are not manyinstances of this message)
Track generation: 1219..1533 -> 314-tiles track
Track generation: 1187..1487 -> 300-tiles track
Track generation: 1177..1475 -> 298-tiles track
Track generation: 1407..1763 -> 356-tiles track
Track generation: 1159..1453 -> 294-tiles track
Track generation: 1112..1398 -> 286-tiles track
retry to generate track (normal if there are not manyinstances of this message)
Track generation: 1296..1624 -> 328-tiles track
Track generation: 1168..1464 -> 296-tiles track
Track generation: 1180..1479 -> 299-tiles track
---------------------------------
| rollout/           |          |
|    ep_len_mean     | 125      |
|    ep_rew_mean     | -58.2    |
| time/              |          |
|    fps             | 57       |
|    iterations      | 1        |
|    time_elapsed    | 35       |
|    total_timesteps | 2048     |
---------------------------------
Using cuda device
Wrapping the env with a `Monitor` wrapper
Wrapping the env in a DummyVecEnv.
Wrapping the env in a VecTransposeImage.
== CURRENT SYSTEM INFO ==
OS: Linux-5.11.0-38-generic-x86_64-with-glibc2.29 #42~20.04.1-Ubuntu SMP Tue Sep 28 20:41:07 UTC 2021
Python: 3.8.10
Stable-Baselines3: 1.3.0
PyTorch: 1.10.0+cu102
GPU Enabled: True
Numpy: 1.20.0
Gym: 0.19.0

== SAVED MODEL SYSTEM INFO ==
OS: Linux-5.11.0-38-generic-x86_64-with-glibc2.29 #42~20.04.1-Ubuntu SMP Tue Sep 28 20:41:07 UTC 2021
Python: 3.8.10
Stable-Baselines3: 1.3.0
PyTorch: 1.10.0+cu102
GPU Enabled: True
Numpy: 1.20.0
Gym: 0.19.0

Track generation: 1184..1484 -> 300-tiles track

Process finished with exit code 0
```

Plot of the actions:
Figure_1

### Expected behavior
Since the two models receive the same input and deterministic=True,
I would expect the resulting actions to be the same.

### System Info
I am using a virtual environment under Ubuntu 20.04 with the following info:
- OS: `Linux-5.11.0-38-generic-x86_64-with-glibc2.29 #42~20.04.1-Ubuntu SMP Tue Sep 28 20:41:07 UTC 2021`
- Stable-Baselines3: 1.3.0
- Python: 3.8.10
- PyTorch: 1.10.0+cu102
- Gym: 0.19.0
- GPU: NVIDIA GeForce RTX 2070
- GPU Enabled: True
- Numpy: 1.20.0

### Additional context
None.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

### PR DESCRIPTION
**Pull request by Demetrio92**
addresses #683, #614

closes #683
closes #614

## Description
Slightly improved the docs for `BaseAlgorithm.load`

## Motivation and Context

Technically, BaseAlgorithm.load is a classmethod, and the docs somewhere mention that loading in-place does not work. However, this behavior might be simply confusing. Especially compared to, for example, tensorflow/keras loading in-place by default. So now it's explicitly explained in the function's doc-string. 

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [] My change requires a change to the documentation.
- [] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

------------------
p.s. tbh, requiring to run an hour of tests for 3 lines of docsting changes is a bit extreme

### CODE REVIEW

**Comment by araffin** [comment]
Hello,
thanks for the PR, i also meant to add a warning here: https://stable-baselines3.readthedocs.io/en/master/guide/examples.html#basic-usage-training-saving-loading

**Comment by Demetrio92** [comment]
@araffin can you allow the runners? 

I think the example is actually doing everything correctly, and anyone who sees it and borrows the code from there should be safe. It's those who just figure out they can `model.load()` get surprised. 

I'll add a note to the example as well.

**Comment by Demetrio92** [comment]
Added a Note to the examples. Interestingly there was something similar there already but was commented out

File: docs/guide/examples.rst [hunk 1]
```diff
@@ -56,10 +56,11 @@ In the following example, we will train, save and load a DQN model on the Lunar
...
   You can install it using ``apt install swig`` and then ``pip install box2d box2d-kengz``
 
-.. .. note::
-..   ``load`` function re-creates model from scratch on each call, which can be slow.
-..   If you need to e.g. evaluate same model with multiple different sets of parameters, consider
-..   using ``load_parameters`` instead.
+.. note::
+  ``load`` method re-creates the model from scratch and should be called on the Algorithm without instantiating it first,
+  e.g. ``model = DQN.load("dqn_lunar", env=env)`` instead of ``model = DQN(env=env)`` followed by  ``model.load("dqn_lunar")``. The latter **will not work** as ``load`` does not work by reference.
+  If you want to load parameters without re-creating the model, e.g. to evaluate same model
```
Comment (Miffyli) [inline docs/guide/examples.rst:62]:
```suggestion
  If you want to load parameters without re-creating the model, e.g. to evaluate the same model 
```

File: docs/guide/examples.rst [hunk 2]
```diff
@@ -56,10 +56,11 @@ In the following example, we will train, save and load a DQN model on the Lunar
...
 
-.. .. note::
-..   ``load`` function re-creates model from scratch on each call, which can be slow.
-..   If you need to e.g. evaluate same model with multiple different sets of parameters, consider
-..   using ``load_parameters`` instead.
+.. note::
+  ``load`` method re-creates the model from scratch and should be called on the Algorithm without instantiating it first,
+  e.g. ``model = DQN.load("dqn_lunar", env=env)`` instead of ``model = DQN(env=env)`` followed by  ``model.load("dqn_lunar")``. The latter **will not work** as ``load`` does not work by reference.
+  If you want to load parameters without re-creating the model, e.g. to evaluate same model 
+  with multiple different sets of parameters, consider using ``load_parameters`` instead.
```
Comment (Miffyli) [inline docs/guide/examples.rst:63]:
```suggestion
  with multiple different sets of parameters, consider using ``set_parameters`` instead.
```

File: docs/misc/changelog.rst [hunk 3]
```diff
@@ -49,6 +49,9 @@ Documentation:
 - Add documentation on exporting to TFLite/Coral
 - Added JMLR paper and updated citation
 - Added link to RL Tips and Tricks video
+- Update BaseAlgorithm.load docstring
```
Comment (Miffyli) [inline docs/misc/changelog.rst:52]:
```suggestion
- Update ``BaseAlgorithm.load`` docstring
```

File: docs/misc/changelog.rst [hunk 4]
```diff
@@ -49,6 +49,9 @@ Documentation:
...
+- Add a Note on `load` behavior in the examples
```
Comment (Miffyli) [inline docs/misc/changelog.rst:53]:
```suggestion
- Add a Note on ``load`` behavior in the examples
```

File: stable_baselines3/common/base_class.py [hunk 5]
```diff
@@ -657,7 +657,9 @@ def load(
         **kwargs,
     ) -> "BaseAlgorithm":
         """
-        Load the model from a zip-file
+        Load the model from a zip-file.
+        Note: `load` re-creates model from scratch, it does not update in-place,
```
Comment (Miffyli) [inline stable_baselines3/common/base_class.py:661]:
```suggestion
        Note: `load` re-creates the model from scratch, it does not update it in-place!
```

File: stable_baselines3/common/base_class.py [hunk 6]
```diff
@@ -657,7 +657,9 @@ def load(
...
+        for an in-place load use `set_parameters` instead.
```
Comment (Miffyli) [inline stable_baselines3/common/base_class.py:662]:
```suggestion
        For an in-place load use `set_parameters` instead.
```

**Review by Miffyli — COMMENTED** [review]
Bunch of in-github-edits for language (did not have write rights to the branch), but other than that LGTM! :)

**Review by Miffyli — APPROVED** [review]
